tests/unit/test_project.py

```diff
@@ -76,139 +112,278 @@ def test_project_unmarshal(yaml_data):
+    mock_platforms = {"arm64": {"build-on": ["arm64"], "build-for": ["amd64"]}}
+    assert "arm64 != amd64" in reload_project_platforms(mock_platforms)
+
+    # Both build and target architectures must be supported
+    mock_platforms = {
+        "mock": {"build-on": ["arm64a", "noarch"], "build-for": ["amd64"]}
+    }
+    assert "none of these build architectures is supported" in reload_project_platforms(
+        mock_platforms
+    )
+
+    mock_platforms = {
+        "mock": {"build-on": ["arm64a", "arm64"], "build-for": ["noarch"]}
+    }
+    assert "build ROCK for target architecture noarch" in reload_project_platforms(
+        mock_platforms
+    )
+
+    # The underlying build machine must be compatible
+    # with both build_on and build_for
+    other_arch = "arm" if BUILD_ON_ARCH == "amd64" else "amd64"
+    mock_platforms = {"mock": {"build-on": [other_arch], "build-for": other_arch}}
+    assert "if the host is compatible with" in reload_project_platforms(mock_platforms)
+
+    mock_platforms = {"mock": {"build-on": [other_arch], "build-for": "amd64"}}
+    assert (
+        f"must be built on one of the following architectures: {[other_arch]}"
+        in reload_project_platforms(mock_platforms)
+    )
+
+
 @pytest.mark.parametrize(
     "field", ["name", "version", "base", "parts", "description", "summary", "license"]
 )
-def test_project_missing_field(yaml_data, field):
-    del yaml_data[field]
+def test_project_missing_field(yaml_loaded_data, field):
+    del yaml_loaded_data[field]
 
     with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_data)
+        Project.unmarshal(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
         f"- field '{field}' required in top-level configuration"
     )
 
 
-def test_project_extra_field(yaml_data):
-    yaml_data["extra"] = "invalid"
+def test_project_extra_field(yaml_loaded_data):
+    yaml_loaded_data["extra"] = "invalid"
 
     with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_data)
+        Project.unmarshal(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
         "- extra field 'extra' not permitted in top-level configuration"
     )
 
 
-def test_project_parts_validation(yaml_data):
-    yaml_data["parts"]["foo"]["invalid"] = True
+def test_project_parts_validation(yaml_loaded_data):
+    yaml_loaded_data["parts"]["foo"]["invalid"] = True
 
     with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_data)
+        Project.unmarshal(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
         "- extra field 'invalid' not permitted in 'parts.foo' configuration"
     )
 
 
-def test_project_load(yaml_data, tmp_path):
-    rockcraft_file = tmp_path / "rockcraft.yaml"
+def test_project_load(yaml_data, yaml_loaded_data):
+    rockcraft_file = Path("rockcraft.yaml")
```

Comment:
Hmm why did you revert this? Did we overwrite each other?
The reason I added the `tmp_path` is that the test was dumping the yaml file on `cwd`, which was visible when e.g. running via `make test-units`.

Reply:
this should have been a merge conflict...I'd need to dig deeper into the Git history to understand why there was no conflict

Reply:
re-added